servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/HeadersPropagatorGetter.java

```diff
@@ -64,7 +65,8 @@ public String get(@Nullable final HttpHeaders carrier, final String key) {
         if (carrier == null) {
             return null;
         }
-        final CharSequence value = carrier.get(key);
+        // For now, we need to normalize the key to lower case for compatibility with our HTTP/2 headers.
+        final CharSequence value = carrier.get(key.toLowerCase(Locale.ENGLISH));
```

Comment:
Capturing original discussion here: https://github.com/apple/servicetalk/pull/3306#discussion_r2241556810

Would you like to prepare a patch for [B3Propagator](https://github.com/open-telemetry/opentelemetry-java/blob/main/extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/B3Propagator.java)?

Reply:
I'll do that as well, but I'd also like to fix it on our side. The case sensitivity of our HTTP/2 headers structure is the problem, the constants used in B3Propagator are just the trigger.